java-frontend/src/main/java/org/sonar/java/se/checks/NullDereferenceCheck.java

```diff
@@ -222,7 +222,9 @@ public ProgramState checkPostStatement(CheckerContext context, Tree syntaxNode)
   }
 
   private static boolean isAnnotatedCheckForNull(MethodInvocationTree syntaxNode) {
-    return syntaxNode.symbol().metadata().isAnnotatedWith("javax.annotation.CheckForNull");
+    return syntaxNode.symbol().metadata().isAnnotatedWith("javax.annotation.CheckForNull")
+      // Despite the name, Spring Nullable is meant to be used as CheckForNull
+      || syntaxNode.symbol().metadata().isAnnotatedWith("org.springframework.lang.Nullable");
```

Comment:
In fact, I think we should rather rely on the `NullnessAnnotationUtils` class here, and move this method there.
There is already a list of nullable annotations, and I believe we should do another pass on the *Nullable* annotations list, to check if some of them are in fact strong nullness constraints (*a la* javax `CheckForNull`), and not weak (*a la* javax `Nullable`), to be used in such context.